docs/features/techdocs/configuration.md

```diff
@@ -70,4 +71,14 @@ techdocs:
       # Cloud Storage Bucket Name
       bucketName: 'techdocs-storage',
 
+    # Required when techdocs.publisher.type is set to 'awsS3'. Skip otherwise.
+
+    awsS3:
+      # An API key is required to write to a storage bucket.
+      credentials:
+        $file: '/path/to/aws_application_credentials.json',
```

Comment:
I agree with you. 
At first I wanted to put the credentials as environment variables:

```yml
credentials:
    aws_access_key_id: ACCESS_KEY_ID
    aws_secret_access_key: SECRET_ACCESS_KEY
```

But to keep some consistency with the configuration for GCS I thought it might be interesting to keep the same credentials file logic. 

But since it is not very relevant for aws we can perhaps replace file credentials by environment variables 👍

@OrkoHunter I would like to know your opinion on that.

Reply:
> Curious, can the credentials be set with just environment variables?

Yes! We can use the contents of the "**credentials.json" as the value for the environment variable (e.g. `TECHDOCS_AWSS3_CREDENTIALS`). And set
```
awsS3:
  credentials:
    $env: TECHDOCS_AWSS3_CREDENTIALS
```

I have used this with GCS. But I see that it will be nice to write this ^ in the `app-config.yaml` comments and the Config reference docs.

Reply:
Wouldn't the environment variable need to be a JSON string in this case? I'm personally not a big fan of managing them that way.

Reply:
@andrewthauer I agree with you that JSON strings are not pretty in environment variables.

Here is my concern. With GCS, their credentials JSON looks like this
```json
{
  "type": "",
  "project_id": "",
  "private_key_id": "",
  "private_key": "-----BEGIN PRIVATE KEY-----<Insert 1600 characters>-----END PRIVATE KEY-----\n",
  "client_email": "",
  "client_id": "",
  "auth_uri": "",
  "token_uri": "",
  "auth_provider_x509_cert_url": "",
  "client_x509_cert_url": ""
}
```

Even if we split these out into individual fields, the `private_key` will still be a huge JSON string. So, I do not see much of an improvement here over the existing ways i.e.
1. Copy over the credentials file as a huge `$env` variable.
2. (Kubernetes) Inject the `$file` using Kubernetes secrets management. https://kubernetes.io/docs/concepts/configuration/secret/#overview-of-secrets
3. ...other typical ways to make a secret file available on a server.

That being said, if AWS allows smaller alphanumeric token-type secrets that can be generated and set as environment variables, I too would prefer that over the huge JSON string. It's just that, GCS is holding us back with how it authenticates service accounts.

Reply:
Ok, in that case, then yeah, I'm not sure it makes sense to mimic GCP for this then. I wonder if these could be a pass through where the child config fields are just turned into an object as is. This would potentially allow you to use both file or individual fields.

For the AWS storage, it might also be nice to support temporary security tokens with STS. This could be done in a subsequent PR, but it would change the config required for security credentials a bit.

Reply:
From what I understand we would have an environment variable `TECHDOCS_AWSS3_CREDENTIALS` with the content of the `credentials.json` stringified but wouldn't it be more relevant to set the credentials individually ?



```yml
awsS3:
  credentials:
    aws_access_key_id:
        $env: TECHDOCS_AWSS3_ACCESS_KEY_ID_CREDENTIAL
    aws_secret_access_key:
        $env: TECHDOCS_AWSS3_SECRET_ACCESS_KEY_CREDENTIAL
```

Reply:
@ayshiff Yes, please go ahead if you think it's an easier process to directly set env variables for S3. I'll be looking forward to the documentation you have written to test the process for myself. Thank you!

Reply:
Actually the keys themselves should be camelCases by convention:

I would suggest:

```yaml
awsS3:
  credentials:
    accessKeyKd:
      $env: TECHDOCS_AWSS3_ACCESS_KEY_ID_CREDENTIAL
    secretAccessKey:
      $env: TECHDOCS_AWSS3_SECRET_ACCESS_KEY_CREDENTIAL
```

To extend this for `STS` in the future it might look something like this:

```yaml
awsS3:
  credentials:
    mode: 'iam-role'
    region:
      $env: AW_S3_REGION
```

Reply:
If I run this as a container in ECS will I be able to skip all of this config and use the role that is assigned to the task?

Reply:
@mfrinnstrom Role and Region are not used as of this moment. Only bucket name and credentials are needed.

Reply:
@OrkoHunter that was my concern. If I add credentials to my task in AWS ECS that would be picked up automatically by the AWS SDK. It seems like we bypass this here and require credentials to be provided from the app.

Reply:
> If I add credentials to my task in AWS ECS

Is this same as setting environment variables? We are open for suggestions. When the AWS sdk is initialized here, should it check some existing environment variables?
May I also ask what are the downsides of explicitly setting those environment variables in `app-config.yaml`?

Reply:
@OrkoHunter not really, although the AWS SDK has some standard environment variables that are picked up automatically if available. Some more information can be found [here](https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/setting-credentials-node.html).

We usually create roles and assign them to our different workloads, in this case an ECS task. This will then be picked up by the AWS SDK in the ECS task without having to create a user and generate credentials for that, and then having to securely manage and supply those credentials to the workload. 

This works great when running in AWS but when running outside of AWS you will need something like what is implemented right now with the access and secret keys as environment variables. It should probably be possible to use the default environment variables (`AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`) and have the SDK pick that up automatically, but that becomes a bit hidden and maybe not to easy to follow in the config.